greenmail-core/src/test/java/com/icegreen/greenmail/test/SendReceiveWithInternationalAddress.java

```diff
@@ -0,0 +1,167 @@
+
+    @Test
+    public void testSend() throws MessagingException, UnsupportedEncodingException {
+
+        Session session = GreenMailUtil.getSession(ServerSetupTest.SMTP, properties);
+        MimeMessage mimeMessage = new InternationalizedMimeMessage(session);
+        mimeMessage.setSubject("subject");
+        mimeMessage.setSentDate(new Date());
+        mimeMessage.setFrom("múchätįldé@tìldę.oœ");
+        mimeMessage.setRecipients(Message.RecipientType.TO, "用户@例子.广告");
+        mimeMessage.setRecipients(Message.RecipientType.CC, "θσερεχα@μπλε.ψομ");
+        mimeMessage.setRecipients(Message.RecipientType.BCC, "राममो@हन.ईन्फो");
+
+        // The body text needs to be encoded if it contains non us-ascii characters
+        mimeMessage.setText(MimeUtility.encodeText("用户@例子"));
+
+        GreenMailUtil.sendMimeMessage(mimeMessage);
+
+        // Decoding the body text to verify equality
+        String decodedText = MimeUtility.decodeText(GreenMailUtil.getBody(greenMail.getReceivedMessages()[0]));
+        assertEquals("用户@例子", decodedText);
+    }
+
+    private static class InternationalizedMimeMessage extends MimeMessage {
+
+        public InternationalizedMimeMessage(Session session) {
+            super(session);
+        }
+
+        // Current Java Mail version does not respect the "mail.mime.address.strict"
+        // value when setting email addresses in a MimeMessage object (calls
+        // InternetAddress.parse(String), which will always validate the
+        // address based on rfc822) so we need to override this method to
+        // make this example work.
+        @Override
+        public void setRecipients(Message.RecipientType type, String addresses) throws MessagingException {
+
+            if (type == RecipientType.NEWSGROUPS) {
+
+                if (addresses == null || addresses.length() == 0)
+                    removeHeader("Newsgroups");
+                else
+                    setHeader("Newsgroups", addresses);
+
+            } else {
+
+                if (!isStrict()) {
+
+                    InternetAddress address = new InternetAddress();
+                    address.setAddress(addresses);
+                    setAddressHeader(getHeaderName(type), new InternetAddress[]{address});
+
+                } else {
+
+                    setAddressHeader(getHeaderName(type),
+
+                            addresses == null ? null : InternetAddress.parse(addresses));
+                }
+            }
+        }
+
+        // Current Java Mail version does not respect the "mail.mime.address.strict"
+        // value when setting email addresses in a MimeMessage object (calls
+        // InternetAddress.parse(String), which will always validate the
+        // address based on rfc822) so we need to override this method to
+        // make this example work.
+        @Override
+        public void setFrom(String address) throws MessagingException {
+
+            if (address == null)
```

Comment:
Same here. To make the test run we only need this:

InternetAddress internetAddress = new InternetAddress();
 internetAddress.setAddress(address);
setAddressHeader("From", new InternetAddress[]{internetAddress});